python/amici/petab_objective.py

```diff
@@ -272,13 +277,25 @@ def create_parameter_mapping(
                      "SBML LocalParameters. If the model contains "
                      "LocalParameters, parameter mapping will fail.")
 
-    prelim_parameter_mapping = \
-        petab_problem.get_optimization_to_simulation_parameter_mapping(
-            warn_unmapped=False, scaled_parameters=scaled_parameters,
-            allow_timepoint_specific_numeric_noise_parameters=
+    default_parameter_mapping_kwargs = {
+        "warn_unmapped": False,
+        "scaled_parameters": scaled_parameters,
+        "allow_timepoint_specific_numeric_noise_parameters":
             not petab.lint.observable_table_has_nontrivial_noise_formula(
-                petab_problem.observable_df
-            )
+                petab_problem.observable_df),
+    }
+    if parameter_mapping_kwargs is None:
+        parameter_mapping_kwargs = {}
+
+    prelim_parameter_mapping = \
+        petab.get_optimization_to_simulation_parameter_mapping(
+            condition_df=petab_problem.condition_df,
+            measurement_df=petab_problem.measurement_df,
+            parameter_df=petab_problem.parameter_df,
+            observable_df=petab_problem.observable_df,
+            sbml_model=petab_problem.sbml_model,
+            **dict(default_parameter_mapping_kwargs,
+                   **parameter_mapping_kwargs)
```

Comment:
Does this work?

Maybe this?
```suggestion
            **{
                **default_parameter_mapping_kwargs,
                **parameter_mapping_kwargs,
            }
```